plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -160,9 +235,12 @@ export class TeamManager {
     private async syncPropertyDefinitions(properties: Properties, team: Team) {
         for (const [key, value] of Object.entries(properties)) {
             if (!this.propertyDefinitionsCache.get(team.id)?.has(key)) {
+                const isNumerical = typeof value === 'number'
+                const { propertyType, propertyTypeFormat } = detectPropertyDefinitionTypes(isNumerical, value, key)
+
                 await this.db.postgresQuery(
-                    `INSERT INTO posthog_propertydefinition (id, name, is_numerical, volume_30_day, query_usage_30_day, team_id) VALUES ($1, $2, $3, NULL, NULL, $4) ON CONFLICT DO NOTHING`,
-                    [new UUIDT().toString(), key, typeof value === 'number', team.id],
+                    `INSERT INTO posthog_propertydefinition (id, name, is_numerical, volume_30_day, query_usage_30_day, team_id, property_type, property_type_format) VALUES ($1, $2, $3, NULL, NULL, $4, $5, $6) ON CONFLICT DO NOTHING`,
+                    [new UUIDT().toString(), key, isNumerical, team.id, propertyType, propertyTypeFormat],
                     'insertPropertyDefinition'
```

Comment:
This will only add the types to new keys, not to any old ones.

Can we update the query to something like this pseudoQL?

```sql
insert bla on conflict if existing.property_type is null then set the darned thing
```